### DISCUSSION TITLE
Ports folder is unwieldy, should be made hierarchical in some way.

### DISCUSSION THREAD

**Discussion by jherico** [microsoft/vcpkg#19068] [Ideas]
The ports folder contains over 1000 subdirectories now.  This starts to become unwieldy in common development UI environments, like Visual Studio Code, or even GitHub:
![image](https://user-images.githubusercontent.com/1533642/61329208-a3d29080-a7d1-11e9-8519-6f4c37fd0f16.png)

The amount of time anyone might spend scrolling through pages of ports in order to find and edit a file in a particular port folder may be trivial individually, but it's a super common and frequent operation.  Suggest that the ports folder be made hierarchical with each port going into a folder named after the first one or two lowercase letters of the port name.  

To future proof it, the vcpkg logic could search for port files incrementally... so foobar port would be looked for first in `ports/foobar`, then in `ports/f/foobar`, then in `ports/f/o/foobar` and so on.

**Comment by BillyONeal** [comment]
Dropping requires-discussion and adding vcpkg-feature; assigning to @strega-nil since I believe she was working on this.

**Comment by strega-nil** [comment]
Fixed by the future implementation of #12881

**Comment by AlexanderTaeschner** [comment]
Why was the issue #7289 closed? The problem with the ports folder having more than 1000 subdirectories still exists, so that more than 700 directories are not accessible via the Github web interface!

**Reply by autoantwort** [reply]
It was closed because the issue was moved to the discussions section. Btw as a workaround you can use `Go to file` and simply enter the name of the port you are searching for or you are using https://vcpkg.io/en/packages.html
